lightblue/src/main/java/org/esbtools/eventhandler/lightblue/BulkLightblueRequester.java

```diff
@@ -0,0 +1,158 @@
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.esbtools.eventhandler.lightblue;
+
+import com.redhat.lightblue.client.LightblueClient;
+import com.redhat.lightblue.client.request.AbstractLightblueDataRequest;
+import com.redhat.lightblue.client.request.DataBulkRequest;
+import com.redhat.lightblue.client.response.LightblueBulkDataResponse;
+import com.redhat.lightblue.client.response.LightblueDataResponse;
+import com.redhat.lightblue.client.response.LightblueErrorResponse;
+import com.redhat.lightblue.client.response.LightblueException;
+import com.redhat.lightblue.client.response.LightblueResponse;
+
+import org.esbtools.eventhandler.Responses;
+import org.esbtools.eventhandler.ResponsesHandler;
+import org.esbtools.eventhandler.Result;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+
+public class BulkLightblueRequester implements LightblueRequester {
+    private final LightblueClient lightblue;
+    private final Map<LazyResult, AbstractLightblueDataRequest[]> queuedRequests =
+            Collections.synchronizedMap(new HashMap<>());
+
+    public BulkLightblueRequester(LightblueClient lightblue) {
+        this.lightblue = lightblue;
+    }
+
+    @Override
+    public LightblueResponsePromise queueRequests(AbstractLightblueDataRequest... requests) {
+        return new BulkResponsePromise(requests);
+    }
+
+    private void doQueuedRequestsAndPopulateResults() throws LightblueException {
+        Map<LazyResult, AbstractLightblueDataRequest[]> batch;
+
+        synchronized (queuedRequests) {
+            batch = new HashMap<>(queuedRequests);
+            queuedRequests.clear();
+        }
+
+        DataBulkRequest bulkRequest = new DataBulkRequest();
+
+        for (AbstractLightblueDataRequest[] allRequestBatches : batch.values()) {
+            for (AbstractLightblueDataRequest requestInBatch : allRequestBatches) {
+                bulkRequest.add(requestInBatch);
+            }
+        }
+
+        LightblueBulkDataResponse bulkResponse = lightblue.bulkData(bulkRequest);
+
+        for (Entry<LazyResult, AbstractLightblueDataRequest[]> lazyResultToRequests : batch.entrySet()) {
+            LazyResult lazyResult = lazyResultToRequests.getKey();
+            AbstractLightblueDataRequest[] requests = lazyResultToRequests.getValue();
+            Map<AbstractLightblueDataRequest, LightblueDataResponse> responseMap = new HashMap<>(requests.length);
+
+            for (AbstractLightblueDataRequest request : requests) {
+                LightblueResponse response = bulkResponse.getResponse(request);
+
+                if (response instanceof LightblueErrorResponse) {
+                    LightblueErrorResponse errorResponse = (LightblueErrorResponse) response;
+                    if (errorResponse.hasLightblueErrors() || errorResponse.hasDataErrors()) {
+                        // TODO: handle errors
```

Comment:
I'm adding in error handling right now that will populate the Result (aka Future) with errors.